firmware/controllers/algo/accel_enrichment.cpp

```diff
@@ -47,34 +48,93 @@ void WallFuel::reset() {
 	memset(wallFuel, 0, sizeof(wallFuel));
 }
 
-floatms_t WallFuel::adjust(int injectorIndex, floatms_t target DECLARE_ENGINE_PARAMETER_SUFFIX) {
-	if (cisnan(target)) {
-		return target;
+floatms_t WallFuel::adjust(int injectorIndex, floatms_t M_des DECLARE_ENGINE_PARAMETER_SUFFIX) {	
+	if (cisnan(M_des)) {
+		return M_des;
 	}
 	// disable this correction for cranking
 	if (ENGINE(rpmCalculator).isCranking(PASS_ENGINE_PARAMETER_SIGNATURE)) {
-		return target;
+		return M_des;
 	}
-	float addedToWallCoef = CONFIG(addedToWallCoef);
 
-	/**
-	 * What amount of fuel is sucked of the walls, based on current amount of fuel on the wall.
-	 */
-	floatms_t suckedOffWallsAmount = wallFuel[injectorIndex] * CONFIG(suckedOffCoef);
+	/*
+		this math is based on 
+				SAE 810494 by C. F. Aquino
+				SAE 1999-01-0553 by Peter J Maloney
 
-	floatms_t adjustedFuelPulse = (target - suckedOffWallsAmount) / (1 - addedToWallCoef);
+		M_cmd = commanded fuel mass (output of this function)
+		M_des = desired fuel mass (input to this function)
+		M_f = fuel film mass (how much is currently on the wall)
 
+		First we compute how much fuel to command, by accounting for
+		a) how much fuel will evaporate from the walls, entering the air
+		b) how much fuel from the injector will hit the walls, being deposited
+
+		Next, we compute how much fuel will be deposited on the walls.  The net
+		effect of these two steps is computed (some leaves walls, some is deposited)
+		and stored back in M_f.
+
+		alpha describes the amount of fuel that REMAINS on the wall per cycle.
+		It is computed as a function of the evaporation time constant (tau) and
+		the time the fuel spent on the wall this cycle, (recriprocal RPM).
+
+		beta describes the amount of fuel that hits the wall.  
+
+		TODO: these parameters, tau and beta vary with various engine parameters,
+		most notably manifold pressure (as a proxy for air speed), and coolant
+		temperature (as a proxy for the intake valve and runner temperature).
+
+		TAU: decreases with increasing temperature.
+		     decreases with decreasing manifold pressure.
+
+		BETA: decreases with increasing temperature.
+		     decreases with decreasing manifold pressure.
+	*/
+
+	// if tau is really small, we get div/0.
+	// you probably meant to disable wwae.
+	float tau = CONFIG(wwaeTau);
+	if(tau < 0.01f)
+	{
+		return M_des;
+	}
+
+	// Ignore really slow RPM
+	int rpm = getRpmE(engine);
+	if(rpm < 100)
+	{
+		return M_des;
+	}
+
+	float alpha = expf_taylor(-120 / (rpm * tau));
+	float beta = CONFIG(wwaeBeta);
+
+	// If beta is larger than alpha, the system is underdamped.
+	// For reasonable values {tau, beta}, this should only be possible
+	// at extremely low engine speeds (<300rpm ish)
+	// Clamp beta to less than alpha.
+	if(beta > alpha)
```

Comment:
one day we will have automated code formatting, for now I will just have to accept your code style I guess